assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -126,10 +126,15 @@ function MediaProvider({ children }) {
   );
 
   const processor = useCallback(
-    ({ mimeType, posterId, id, src, videoId }) => {
+    ({ mimeType, posterId, src, videoId, local }) => {
       const process = async () => {
-        if (allowedVideoMimeTypes.includes(mimeType) && !posterId && videoId) {
-          await uploadVideoPoster(id, src);
+        if (
+          allowedVideoMimeTypes.includes(mimeType) &&
+          !local &&
+          !posterId &&
+          videoId
+        ) {
+          await uploadVideoPoster(videoId, src);
```

Comment:
How will the `local == true` poster be updated? At the upload time we know both the video content and poster and can submit them at about the same time (even if one has to follow the other closely).

Reply:
Closing this one in favor of https://github.com/google/web-stories-wp/pull/1037